### PR TITLE
Use FF to capture all Gong transcripts

### PR DESCRIPTION
**Pull request by albandum**
## Description

- Use FF to capture all Gong transcripts
- Let only one user set up the default storage for everyone when the FF is set on a workspace. Show a message about it.
Screenshot 2024-11-19 at 15 18 13
- Add isDefaultFullStorage attribute to labs transcripts configuration

## Deploy Plan

Run migration 115

### CODE REVIEW

File: front/lib/resources/labs_transcripts_resource.ts [hunk 1]
```diff
@@ -155,6 +156,29 @@ export class LabsTranscriptsConfigurationResource extends BaseResource<LabsTrans
     return this.update({ dataSourceViewId: dataSourceView.id });
   }
 
+  static async fetchFirstConfigurationWithDatasourceViewForWorkspace(
+    auth: Authenticator
+  ): Promise<LabsTranscriptsConfigurationResource | null> {
+    const configuration = await LabsTranscriptsConfigurationModel.findOne({
+      where: {
+        workspaceId: auth.workspace()?.id,
```
Comment (flvndvd) [inline front/lib/resources/labs_transcripts_resource.ts:164]:
```suggestion
        workspaceId: auth.getNonNullableWorkspace(),
```

File: front/temporal/labs/activities.ts [hunk 2]
```diff
@@ -268,8 +269,63 @@ export async function processTranscriptActivity(
...
+        await LabsTranscriptsConfigurationResource.fetchFirstConfigurationWithDatasourceViewForWorkspace(
+          auth
+        );
+
+      if (!defaultTranscriptConfiguration) {
+        localLogger.error(
+          {},
+          "[processTranscriptActivity] No default transcript configuration with a datasource view found while full Gong storage if enabled. Stopping."
+        );
+        await stopRetrieveTranscriptsWorkflow(transcriptsConfiguration);
```
Comment (flvndvd) [inline front/temporal/labs/activities.ts:292]: Wait should this be blocking the whole transcript workflow?

File: front/temporal/labs/activities.ts [hunk 3]
```diff
@@ -268,8 +269,63 @@ export async function processTranscriptActivity(
     return;
   }
 
+  // labs_transcripts_gong_full_storage FF enables storing all Gong transcripts in a single datasource view
+  let gongFullStorage = false;
```
Comment (flvndvd) [inline front/temporal/labs/activities.ts:273]: If I'm reading the code correctly, we won't reach this part if the user was not a participant in the call, right?

Comment (albandum) [reply front/temporal/labs/activities.ts:273]: You're right, forgot about that part. Will dig.

**Review by flvndvd — APPROVED** [review]
Good enough to unblock the situation. Left a few nits.

File: front/temporal/labs/activities.ts (hunk 2 above)
Comment (albandum) [reply front/temporal/labs/activities.ts:281]: Ok, not stopping workflow indeed

**Review by flvndvd — APPROVED** [review]
✅

**Pull request merged**